sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessorClient.cs

```diff
@@ -292,67 +292,67 @@ protected ServiceBusProcessorClient()
         /// <param name="propertiesToModify">The properties of the message to modify while moving to sub-queue.</param>
         /// <param name="cancellationToken"></param>
         /// <remarks>
-        /// A lock token can be found in <see cref="ServiceBusMessage.SystemPropertiesCollection.LockToken"/>,
+        /// A lock token can be found in <see cref="ServiceBusReceivedMessage.LockToken"/>,
         /// only when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
         /// In order to receive a message from the deadletter queue, you will need a new IMessageReceiver"/>, with the corresponding path.
         /// You can use EntityNameHelper.FormatDeadLetterPath(string)"/> to help with this.
         /// This operation can only be performed on messages that were received by this receiver.
         /// </remarks>
-        public virtual async Task DeadLetterAsync(string lockToken, IDictionary<string, object> propertiesToModify = null, CancellationToken cancellationToken = default)
+        public virtual async Task DeadLetterAsync(ServiceBusReceivedMessage message, IDictionary<string, object> propertiesToModify = null, CancellationToken cancellationToken = default)
         {
             // TODO implement
 
             await Task.Delay(1).ConfigureAwait(false);
-            throw new NotImplementedException();
+            //throw new NotImplementedException();
         }
 
         /// <summary>
         /// Moves a message to the deadletter sub-queue.
         /// </summary>
-        /// <param name="lockToken">The lock token of the corresponding message to deadletter.</param>
+        /// <param name="message">The lock token of the corresponding message to deadletter.</param>
         /// <param name="deadLetterReason">The reason for deadlettering the message.</param>
         /// <param name="deadLetterErrorDescription">The error description for deadlettering the message.</param>
         /// <param name="cancellationToken"></param>
         /// <remarks>
-        /// A lock token can be found in <see cref="ServiceBusMessage.SystemPropertiesCollection.LockToken"/>,
+        /// A lock token can be found in <see cref="ServiceBusReceivedMessage.LockToken"/>,
         /// only when <see cref="ReceiveMode"/> is set to <see cref="ReceiveMode.PeekLock"/>.
         /// In order to receive a message from the deadletter queue, you will need a new <see cref="ServiceBusProcessorClient"/>, with the corresponding path.
         /// You can use EntityNameHelper.FormatDeadLetterPath(string) to help with this.
         /// This operation can only be performed on messages that were received by this receiver.
         /// </remarks>
-        public virtual async Task DeadLetterAsync(string lockToken, string deadLetterReason, string deadLetterErrorDescription = null,
+        public virtual async Task DeadLetterAsync(ServiceBusReceivedMessage message, string deadLetterReason, string deadLetterErrorDescription = null,
```

Comment:
Out of curiosity, what's the motivation to pass the entire message rather than lock token?

Reply:
We feel that it is more intuitive and consistent across APIs to always take the message object.
It seemed like the lockToken was partly used to cover up for the fact that we had one message type for both send and receive. With the split to have two separate types, this seems less important.

Reply:
I'd verify that with Track1 designers rather than assuming 🙂
@nemakam was there any additional consideration why would lock token be passed rather than the entire `Message`? Thank you.

Reply:
Yep we are meeting with Track 1 team this week to go over current API, but early feedback is always better 😀

Reply:
I should change my GH name to `SeanFeldman-oracle`.